### DISCUSSION TITLE
Support bootstrapping with files

### DISCUSSION THREAD

**Discussion by bigkevmcd** [fluxcd/flux2#2759] [Proposals]
In order to bootstrap with a pre-configured workload, it'd be useful if we could do something like:

```shell
$ flux bootstrap ... --with-resource-file https://example.com/my-file.yaml
```

Where the expected behaviour would be to read the file and write it to the repo when it's being bootstrapped, in such a way that by default, the file would be sourced by the default `Kustomization` resource.

It shouldn't matter what's in the file, the onus is on the user to provide a valid set of resources.

**Comment by wibed** [comment]
it also separates fetches from within 2 different environments. 
first on the computer it is run on.
and second from within the container flux operates.

this makes it difficult if someone would like to spin up a cluster flux manages from a git server located within the same cluster.
